### ISSUE TITLE
Saving credentials to external storage failed

### ISSUE THREAD

**Issue by RobertoLayna** [pushy-me/pushy-demo-android#12]
all permissions granted, works in android 9 and below.

**Comment by pushy** [comment]
Hi @RobertoLayna,
We'd be glad to assist.

Are you requesting from the user to grant access to `WRITE_EXTERNAL_STORAGE` via a permission dialog, before calling `Pushy.register()`?

```
// Check whether the user has granted us the READ/WRITE_EXTERNAL_STORAGE permissions
if (ContextCompat.checkSelfPermission(getApplicationContext(), Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
    // Request both READ_EXTERNAL_STORAGE and WRITE_EXTERNAL_STORAGE so that the
    // Pushy SDK will be able to persist the device token in the external storage
    ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.READ_EXTERNAL_STORAGE, Manifest.permission.WRITE_EXTERNAL_STORAGE}, 0);
}
```

If the user doesn't consent, the SDK won't be able to write to the external storage.

**Comment by RobertoLayna** [comment]
I require them before calling the register,
I have tested it in versions prior to android 10 and it works without any problem ( checking the path where they are saved )
- Android 10 
- Moto g(7) power

**Comment by pushy** [comment]
Hi @RobertoLayna,
Can you please let us know your app's `targertSdkVersion` defined in `app/build.gradle`?

**Comment by RobertoLayna** [comment]
targetSdkVersion 30
implementation 'me.pushy:sdk:1.0.56'

Now tested on:
- Huawei P20
- Android 11

**Comment by pushy** [comment]
This is most likely due to the enforced [scoped storage](https://developer.android.com/about/versions/10/privacy/changes#scoped-storage) in apps targeting Android 10 and higher.

We've just released version `1.0.58` of the [Pushy Android SDK](https://pushy.me/docs/android/get-sdk) which drops the requirement for storage access completely. 

The SDK now uses the device's [`ANDROID_ID`](https://developer.android.com/reference/android/provider/Settings.Secure#ANDROID_ID), in combination with the device manufacturer and model name, for unique device recognition between app reinstallations. 

In addition, the `android.permission.WRITE_EXTERNAL_STORAGE` permission is no longer required by the SDK. You may remove it if your app does not require it.

Please import version `1.0.58` of the SDK as a dependency in your app's `build.gradle`:

```
implementation 'me.pushy:sdk:1.0.58'
```

Run your app, and let us know if everything works as expected.
